napari/_tests/test_examples.py

```diff
@@ -63,4 +63,9 @@ def raise_errors(etype, value, tb):
     monkeypatch.setattr(notification_manager, 'receive_error', raise_errors)
 
     # run the example!
-    runpy.run_path(str(EXAMPLE_DIR / fname))
+    try:
+        runpy.run_path(str(EXAMPLE_DIR / fname))
+    except SystemExit as e:
+        # we use sys.exit(0) to gracefully exit from examples
+        if e.code != 0:
+            raise
```

Comment:
Should we re-raise the original exception?

```suggestion
            raise e
```

Reply:
A plain `raise` works the same. Actually, apparently there's a small difference. Using the explicit form, it adds one more thing (frame?) to the stack:
```python
In [6]: try:
   ...:     raise Exception('thing')
   ...: except Exception as e:
   ...:     raise e
   ...:
---------------------------------------------------------------------------
Exception                                 Traceback (most recent call last)
<ipython-input-6-cffc773d8242> in <module>
      2     raise Exception('thing')
      3 except Exception as e:
----> 4     raise e
      5

<ipython-input-6-cffc773d8242> in <module>
      1 try:
----> 2     raise Exception('thing')
      3 except Exception as e:
      4     raise e
      5

Exception: thing

In [7]: try:
   ...:     raise Exception('thing')
   ...: except Exception as e:
   ...:     raise
   ...:
---------------------------------------------------------------------------
Exception                                 Traceback (most recent call last)
<ipython-input-7-a71bce9afec2> in <module>
      1 try:
----> 2     raise Exception('thing')
      3 except Exception as e:
      4     raise
      5

Exception: thing
```

Reply:
TIL. Thanks!